kernel/src/scan/mod.rs

```diff
@@ -258,6 +259,35 @@ impl<'a> ExpressionTransform<'a> for ApplyColumnMappings {
     }
 }
 
+static RESTORED_ADD_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    use crate::scan::log_replay::DEFAULT_ROW_COMMIT_VERSION_NAME;
+
+    let partition_values = MapType::new(DataType::STRING, DataType::STRING, true);
+    StructType::new_unchecked(vec![StructField::nullable(
+        "add",
+        StructType::new_unchecked(vec![
+            StructField::not_null("path", DataType::STRING),
+            StructField::not_null("partitionValues", partition_values),
+            StructField::not_null("size", DataType::LONG),
+            StructField::nullable("modificationTime", DataType::LONG),
+            StructField::nullable("stats", DataType::STRING),
+            StructField::nullable(
```

Comment:
looks like it got this `tags` field to, and DEFAULT_ROW_COMMIT_VERSION_NAME. I assume that's because we need to communicate that all into the txn when we re-write new actions for the file?

Reply:
Yes, but I believe these were added in the previous PR (remove files), you can see these in the removed lines below, I think the only net new one is cluster provider name (which is needed for Add action but not remove).

Reply:
these fields where added in the stacked PR (remove_files) clustering provider is the net new one.